### PR TITLE
add integ test for `istioctl analyze` timeout

### PR DESCRIPTION
**Pull request by sushicw**
Add an integration test case for `istioctl analyze --timeout`. Followup to https://github.com/istio/istio/pull/19890

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by sushicw** [comment]
> why does this need to be an integration test be that is deploying kubernetes, etc, etc just to run a single command that times out immediately? it seems like it could just be a unit test? but maybe I am misunderstanding the impl

That's a fair criticism, this did end up being a bit of overkill. I was going for a quick implementation that takes advantage of the existing test suite & framework components for testing `istioctl`. It was easier than setting up more narrowly scoped test scaffolding that doesn't currently exist: that didn't seem worth it for just this test. 

The mitigating factor here is that this isn't a whole new test suite, just a new test, so all the expensive setup has already been done. We're only adding a fraction of a second to the test execution.

**Comment by sushicw** [comment]
> I think there is some overhead from the framework
> 
> ```
> === RUN   TestTimeout
> 2020-01-10T17:30:17.891797Z	info	CI	=== BEGIN: Test: 'galley_test[TestTimeout]' ===
> 2020-01-10T17:30:28.713643Z	info	CI	=== DONE:  Test: 'galley_test[TestTimeout] (10.82184454s)' ===
> --- PASS: TestTimeout (10.83s)
> ```
> 
> That being said, 10s is not bad

That's what I see when I run with `--test.run TestTimeout`. When I run the whole galley suite, TestTimeout takes less than 0.1 seconds.

**Comment by howardjohn** [comment]
That 10s was from the CI. Anyhow I really don't care too much about 10s

**Comment by sushicw** [comment]
Putting on hold pending https://github.com/istio/istio/issues/20072

File: tests/integration/galley/analyze_test.go [hunk 1]
```diff
@@ -180,6 +180,25 @@ func TestAllNamespaces(t *testing.T) {
...
+
+			ns := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-analyze",
+				Inject: true,
+			})
+
+			istioCtl := istioctl.NewOrFail(t, ctx, istioctl.Config{})
+
+			// We should time out immediately.
+			_, err := istioctlSafe(t, istioCtl, ns.Name(), true, "--timeout=0s")
```
Comment (ayj) [inline tests/integration/galley/analyze_test.go]: I would expect `0s` to disable the timeout?

Comment (sushicw) [reply tests/integration/galley/analyze_test.go]: It doesn't, although perhaps it should. It's something I considered briefly but didn't implement. I wasn't sure that "set the timeout to 0 to disable it" would be any more intuitive to users than "set the timeout to an absurdly high number to (effectively) disable it."

**Pull request merged**